common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -304,7 +304,7 @@ public static void throwException(Throwable t) {
   static {
     boolean _unaligned;
     String arch = System.getProperty("os.arch", "");
-    if (arch.equals("ppc64le") || arch.equals("ppc64")) {
+    if (arch.equals("ppc64le") || arch.equals("ppc64") || arch.matches("s390x")) {
```

Comment:
Still doesn't need `.matches`; that's not a regex

Reply:
@srowen : I had used "arch.equals("s390x") initially and by reading ur earlier comments i understood you asking me to use "matches", that's what i did in this new PR

Do you stilll suggest to change it to some thing else?

Reply:
I'm suggesting `.contains` for substring matching. `.matches` is for a regex.

Reply:
> I'm suggesting `.contains` for substring matching. `.matches` is for a regex.

so should make the file change in this PR itself or do  a new PR?

Reply:
Just change it here, push a new commit.

Reply:
done